pkg/lib/oauth/login_hint.go

```diff
@@ -33,8 +35,35 @@ type LoginHint struct {
 	LoginIDPhone    string
 }
 
+func (h *LoginHint) String() string {
+	q := url.Values{}
+	switch h.Type {
+	case LoginHintTypeLoginID:
+		q.Set("type", string(LoginHintTypeLoginID))
+		q.Set("enforce", strconv.FormatBool(h.Enforce))
+		if h.LoginIDEmail != "" {
+			q.Set("email", h.LoginIDEmail)
+		}
+		if h.LoginIDPhone != "" {
+			q.Set("phone", h.LoginIDPhone)
+		}
+		if h.LoginIDUsername != "" {
+			q.Set("username", h.LoginIDUsername)
+		}
+	default:
+		panic(fmt.Errorf("cannot convert login_hint to string with type: %v", h.Type))
```

Comment:
This String() method should support other type as well. Add test for this method.